net/http/src/THttpServer.cxx

```diff
@@ -181,15 +181,20 @@ THttpServer::THttpServer(const char *engine) : TNamed("http", "ROOT http server"
       }
    }
 
-   AddLocation("currentdir/", ".");
+   Bool_t basic_sniffer = strstr(engine, "basic_sniffer") != nullptr;
```

Comment:
Can an attacker set the engine to `basic_sniffer`, thus opening access to `./`?

Reply:
Basic sniffer is not engine like civetweb or fastcgi.
Sniffer is object which can access to some global lists or to registered objects.
Basic sniffer do not that much - only keep list of registered widgets and redirects requests to them by URL.

Reply:
Not the point. I can have a mean html request that uses a forged `basic_sniffer` as the engine. What does that allow me to do, which expectations of the code is that going to violate?

Reply:
In the THttpServer there is a code:
```
   if (strstr(engine, "basic_sniffer")) {
      sniff = new TRootSniffer("sniff");
      sniff->SetScanGlobalDir(kFALSE);
      sniff->CreateOwnTopFolder(); // use dedicated folder
   } else {
      sniff = (TRootSniffer *)gROOT->ProcessLineSync("new TRootSnifferFull(\"sniff\");");
   }
```

basic sniffer means - it only can access registered objects in dedicated folder. 
Global lists are not scanned and cannot be accessed.

In case of web gui only instances of `THttpWSHandler` (web sockets handler) registered to the sniffer. 
Each `RWebWindow` registers own web sockets handler and handles incoming requests.

JSROOT/openui5 files serving done directly with THttpServer functionality.